### PR TITLE
Add a geolift example to the docs

### PR DESCRIPTION
**Pull request by drbenvincent**
Addresses #134

* Added code to generate synthetic data. This is kind of cool because it generates seasonality by sampling from a periodic Gaussian process.
* Dataset saved and added in to repo.
* Added a notebook demonstrating geolift + added it into the docs.
* Added that example as an integration test.

I'm submitting this PR now because I'd benefit from some feedback. I think it does need some work to be a _really_ awesome showcase. Do feel free to raise the bar... I'm a sometimes a bit too trigger happy, releasing examples while they are "good" but not yet "fantastic".

### CODE REVIEW

**Comment by drbenvincent** [comment]
There should probably be some mention of this in the `README.md` and `index.rst`

File: docs/notebooks/geolift1.ipynb [hunk 1]
```diff
@@ -0,0 +1,614 @@
+{
```
Comment (twiecki) [inline docs/notebooks/geolift1.ipynb:1]: <p>company want<strong>s</strong> you to</p><p>uplify -&gt; uplift</p><p> </p><p>Love the narrative and the stakes in this intro!</p><p> </p><p>It turns out that <code>CausalPy</code> does exactly this. -&gt; You find a library called <code>CausalPy</code> that supports exactly that use case and are delighted.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/CausalPy/pull/135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (twiecki) [inline docs/notebooks/geolift1.ipynb:1]: <p>Maybe just a single sentence what we're seeing and what the values represent.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/CausalPy/pull/135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (twiecki) [inline docs/notebooks/geolift1.ipynb:1]: <p>turn progress bar off.</p><p> </p><p>Also, briefly explain what this is doing, maybe with some references to other docs for more info.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/CausalPy/pull/135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (drbenvincent) [reply docs/notebooks/geolift1.ipynb:1]: <p>Done - in an upcoming commit.</p>

Comment (drbenvincent) [reply docs/notebooks/geolift1.ipynb:1]: <p>Done</p>

**Comment by drbenvincent** [comment]
Ok. Whole bunch of updates to the notebook done. Also silenced the progress bar for posterior predictive sampling, so there's less clutter in the notebook.

**Comment by twiecki** [comment]
That's much better, should be in fact be a blog post!

**Pull request merged**